### ISSUE TITLE
[BUG][Fuzzer][wrong-code] `T.ceildiv` silently returns the wrong result for a negative divisor (e.g. `ceildiv(7, -3)` gives `-1` instead of `-2`)

### ISSUE THREAD

**Issue by Hughshine** [tile-ai/tilelang#2943]
### Required prerequisites

- [x] I have read the documentation <https://tilelang.com>.
- [x] I have searched the [Issue Tracker](https://github.com/tile-ai/tilelang/issues) and found no existing report of this defect.

### What version of TileLang are you using?

0.1.13 (latest release)

### System information

TileLang 0.1.13, PyTorch 2.13 / CUDA 13.0, NVIDIA L40S (sm_89). The defect is in host-side IR construction (the arithmetic identity `ceildiv` lowers to), so it is architecture-independent — codegen faithfully computes the wrong expression on any target.

### Problem description

`T.ceildiv(a, b)` on `int32` operands returns a value that is not `ceil(a / b)` whenever the divisor `b` is negative. For example `ceildiv(7, -3)` returns `-1`, but `ceil(7 / -3) = ceil(-2.333…) = -2`. The result is silently wrong — the kernel compiles and runs with no error. Positive divisors are correct.

`ceildiv` is defined as the identity `floordiv(a + b - 1, b)`, which computes the ceiling of `a / b` only when `b > 0`. For `b < 0` the `+ b - 1` bias is applied in the wrong direction, so the result is off (and for divisible cases it lands a full step away). The same wrong value appears on both paths: the runtime (device codegen) path and the compile-time constant-folded path.

<details><summary>Trigger boundary — sign of the divisor (const-folded, from the repro's inline oracle)</summary>

| `a` | `b` | `T.ceildiv(a,b)` | `ceil(a/b)` |
|---|---|---|---|
| 7 | 3 | 3 | 3 (correct — b>0) |
| 7 | -3 | -1 | **-2** |
| 7 | -7 | 0 | **-1** |
| 32 | -2 | -15 | **-16** |
| -31 | -2 | 17 | **16** |
| -7 | 2 | -3 | -3 (correct — b>0) |
| -7 | -2 | 5 | **4** |

Every wrong row has `b < 0`; every `b > 0` row is correct.

</details>

Not a regression — see Provenance.

### Reproducible example code

```python
import torch, tilelang, tilelang.language as T
from fractions import Fraction
from math import ceil

@tilelang.jit(out_idx=[-1])
def make():
    @T.prim_func
    def main(A: T.Tensor((4,), "int32"), B: T.Tensor((4,), "int32"),
             C: T.Tensor((4,), "int32")):
        with T.Kernel(1, threads=4) as _:
            i = T.get_thread_binding()
            C[i] = T.ceildiv(A[i], B[i])
    return main

k = make()
a = torch.tensor([7,   7,  32, -31], dtype=torch.int32, device="cuda")

# negative divisors -> silently wrong
b = torch.tensor([-3, -7,  -2,  -2], dtype=torch.int32, device="cuda")
got = k(a, b).cpu().tolist()
exp = [ceil(Fraction(int(x), int(y))) for x, y in zip(a.tolist(), b.tolist())]
print("neg divisor:", got, "expected", exp)   # -> [-1, 0, -15, 17]  expected [-2, -1, -16, 16]

# same kernel, positive divisors -> correct (control)
bp = torch.tensor([3, 7, 2, 2], dtype=torch.int32, device="cuda")
gotp = k(a, bp).cpu().tolist()
expp = [ceil(Fraction(int(x), int(y))) for x, y in zip(a.tolist(), bp.tolist())]
print("pos divisor:", gotp, "expected", expp)  # -> [3, 1, 16, -15]  matches (correct)
```

### Traceback

No traceback — the kernel compiles and runs to completion; the result is silently wrong and deterministic.

### Expected behavior

`ceildiv(a, b)` should equal `ceil(a / b)` (true-division ceiling) for all integer operands, matching `math.ceil(Fraction(a, b))` and Python's `-(-a // b)` — including negative divisors. Concretely `ceildiv(7, -3) == -2`.

### Additional context

**Root cause.** `ceildiv` lowers to an arithmetic identity that is only valid for a positive divisor. It is defined as `floordiv(a + b - 1, b)` at [`src/tir/op/op.cc:589`](https://github.com/TileLang/tvm/blob/0e15b274bce8b46f971abf5ac390e844aa6acee5/src/tir/op/op.cc#L589) (with the constant-folding twin at [L588](https://github.com/TileLang/tvm/blob/0e15b274bce8b46f971abf5ac390e844aa6acee5/src/tir/op/op.cc#L588)). The `+ b - 1` term biases the numerator up by `(b - 1)` so that floor-division rounds toward `+∞`; this is the standard trick, but it holds only when `b > 0`. When `b < 0` the bias points the wrong way (and `floordiv` itself rounds toward `-∞`), so the ceiling is off by up to one full step.

**Suggested fix.** Replace the `floordiv(a + b - 1, b)` identity at `op.cc:589` (and its const-fold twin at L588) with a sign-correct integer ceildiv that does not assume `b > 0` — e.g. guard the current bias to `b > 0` and use the mirrored bias for `b < 0`, or compute via truncating division plus a sign-aware correction. This touches both the const-fold and runtime branches (not verified end-to-end).

**Provenance.** The `floordiv(a + b - 1, b)` definition of `ceildiv` is long-standing — it predates the TileLang fork and traces to upstream apache/tvm ([#9996](https://github.com/apache/tvm/pull/9996), `git log -S`). Not a regression; wrong since the helper was written.

**Dedup.** I searched the open and closed tracker (`ceildiv` in title/body) and found no existing report of the negative-divisor defect; the only `ceildiv` hits are unrelated (#1856 a `cdiv` alias, #210 a `K // block_K → ceildiv` refactor).

**Reach.** `T.ceildiv` is a core, ubiquitous op — 490 call sites across `examples/` (grid/loop-bound computation, e.g. `T.ceildiv(N, block_N)`, `T.ceildiv(K, block_K)`). The rare ingredient is a **negative divisor**: every shipped `examples/`/`testing/` use passes a positive tile/extent divisor (grepped this session — all `ceildiv(dim, block)` forms with positive `dim`/`block`), so none trip the bug and CI stays green. A negative divisor is not exercised by any example or test; the input is legitimate per the op's contract — the docstring describes a "Generic ceildiv operator" over integers with no sign restriction, and the frontend accepts any `int`/`uint` operand.
